### PR TITLE
Adds tracing sample for cloud spanner

### PR DESCRIPTION
**Pull request by vkedia**
cc @bogdandrutu

### CODE REVIEW

File: spanner/cloud-client/pom.xml [hunk 1]
```diff
@@ -66,6 +70,31 @@ limitations under the License.
       <artifactId>guava</artifactId>
       <version>20.0</version>
     </dependency>
+    <dependency>
+        <groupId>io.opencensus</groupId>
+        <artifactId>opencensus-api</artifactId>
+        <version>0.11.0</version>
```
Comment (bogdandrutu) [inline spanner/cloud-client/pom.xml]: Make it a variable because it is used in few places?

File: spanner/cloud-client/pom.xml [hunk 2]
```diff
@@ -66,6 +70,31 @@ limitations under the License.
...
+    </dependency>
+    <dependency>
+        <groupId>io.opencensus</groupId>
+        <artifactId>opencensus-impl</artifactId>
+        <version>0.11.0</version>
```
Comment (bogdandrutu) [inline spanner/cloud-client/pom.xml]:
```
<scope>runtime</scope>
```

File: spanner/cloud-client/src/main/java/com/example/spanner/TracingSample.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+      return;
+    }
+    SpannerOptions options = SpannerOptions.newBuilder().build();
+    Spanner spanner = options.getService();
+
+    // Installs a handler for /tracez page.
+    ZPageHandlers.startHttpServerAndRegisterAll(8080);
+    // Installs an exporter for stack driver.
+    StackdriverExporter.createAndRegister();
+    Tracing.getExportComponent().getSampledSpanStore().registerSpanNamesForCollection(Arrays.asList("CloudSpannerSample"));
```
Comment (bogdandrutu) [inline spanner/cloud-client/src/main/java/com/example/spanner/TracingSample.java]: private static final String CLOUD_SPANNER_SAMPLE_SPAN_NAME = "CloudSpannerSample";

Not necessary the long suggested variable name but that will ensure that the string is the same here and 10 lines below.

**Comment by bogdandrutu** [comment]
Also you can install the StackdriverStatsExporter as well to push grpc stats to monitoring, here are instructions:
https://github.com/census-instrumentation/opencensus-java/tree/master/exporters/stats/stackdriver

**Comment by lesv** [comment]
You might wish to add something about this to the README.md

**Comment by lesv** [comment]
@rakyll Once they make the changes you've asked for, add the `kokoro-forcerun` label before accepting or merging.

**Comment by bogdandrutu** [comment]
If you link the stats you also need to define the views (or what data to export), to do that do the following:
* Add a compilation dependency on io.opencensus:opencensus-contrib-grpc-metrics:0.11.0
* In the main function call io.opencensus.contrib.grpc.metrics.RpcViews.registerAllCumulativeViews()

**Comment by vkedia** [comment]
Addressed all the comments. PTALA.
I am getting an exception while exporting stats:
```
Jan 23, 2018 9:30:04 AM io.opencensus.exporter.stats.stackdriver.StackdriverExporterWorker export
WARNING: ApiException thrown when exporting TimeSeries.
com.google.api.gax.rpc.UnavailableException: io.grpc.StatusRuntimeException: UNAVAILABLE: io exception
	at com.google.api.gax.rpc.ApiExceptionFactory.createException(ApiExceptionFactory.java:69)
	at com.google.api.gax.grpc.GrpcExceptionCallable$ExceptionTransformingFuture.setException(GrpcExceptionCallable.java:118)
	at com.google.api.gax.grpc.GrpcExceptionCallable$ExceptionTransformingFuture.onFailure(GrpcExceptionCallable.java:101)
	at com.google.api.core.ApiFutures$1.onFailure(ApiFutures.java:61)
	at com.google.common.util.concurrent.Futures$4.run(Futures.java:1123)
	at com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute(MoreExecutors.java:435)
	at com.google.common.util.concurrent.AbstractFuture.executeListener(AbstractFuture.java:900)
	at com.google.common.util.concurrent.AbstractFuture.complete(AbstractFuture.java:811)
	at com.google.common.util.concurrent.AbstractFuture.setException(AbstractFuture.java:675)
	at io.grpc.stub.ClientCalls$GrpcFuture.setException(ClientCalls.java:492)
	at io.grpc.stub.ClientCalls$UnaryStreamToFuture.onClose(ClientCalls.java:467)
	at io.grpc.ForwardingClientCallListener.onClose(ForwardingClientCallListener.java:41)
	at io.grpc.internal.CensusStatsModule$StatsClientInterceptor$1$1.onClose(CensusStatsModule.java:684)
	at io.grpc.ForwardingClientCallListener.onClose(ForwardingClientCallListener.java:41)
	at io.grpc.internal.CensusTracingModule$TracingClientInterceptor$1$1.onClose(CensusTracingModule.java:392)
	at io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.java:475)
	at io.grpc.internal.ClientCallImpl.access$300(ClientCallImpl.java:63)
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl.close(ClientCallImpl.java:557)
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl.access$600(ClientCallImpl.java:478)
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInContext(ClientCallImpl.java:590)
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:123)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: io.grpc.StatusRuntimeException: UNAVAILABLE: io exception
	at io.grpc.Status.asRuntimeException(Status.java:526)
	... 19 more
Caused by: java.io.IOException: Operation timed out
	at sun.nio.ch.FileDispatcherImpl.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:39)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:192)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380)
	at io.netty.buffer.PooledUnsafeDirectByteBuf.setBytes(PooledUnsafeDirectByteBuf.java:288)
	at io.netty.buffer.AbstractByteBuf.writeBytes(AbstractByteBuf.java:1106)
	at io.netty.channel.socket.nio.NioSocketChannel.doReadBytes(NioSocketChannel.java:343)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:123)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:645)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:580)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:497)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:459)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:858)
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:138)
	... 1 more
```

**Comment by vkedia** [comment]
Enabling SD premium made it work.

**Comment by vkedia** [comment]
@lesv Can you please approve.

**Comment by lesv** [comment]
Waiting on Kokoro.

**Comment by vkedia** [comment]
Thanks!

**Pull request merged**